app/src/lib/editors/win32.ts

```diff
@@ -82,6 +86,54 @@ const LocalMachineUninstallKey = (subKey: string) =>
 const Wow64LocalMachineUninstallKey = (subKey: string) =>
   registryKey(HKEY.HKEY_LOCAL_MACHINE, wow64UninstallSubKey, subKey)
 
+
+/* 
+* This function generates registry keys for a given Visual Studio product
+* Currently, we only support 64 bit versions of Visual Studio
+*
+* To find the product ID, please consult the following link:
+* https://docs.microsoft.com/en-us/visualstudio/install/workload-and-component-ids?view=vs-2022 
+* (change the year at the end of the url to the year you want)
+* 
+* To find the version, please consult the following link:
+* https://docs.microsoft.com/en-us/visualstudio/install/visual-studio-build-numbers-and-release-dates?view=vs-2022
+* (change the year at the end of the url to the year you want)
+*/
+const registryKeysForVisualStudioIDE = (
+  version: number,
+  productId: string
+): ReadonlyArray<RegistryKey> => {
+
+  const result = new Array<RegistryKey>()
+  /*
+  * vswhere is a tool that is part of the Visual Studio Installer. It is used
+  * to find the path to the Visual Studio installation. 
+  * vswhere is included with the installer as of Visual Studio 2017 version 15.2 and later
+  * https://github.com/microsoft/vswhere
+  * 
+  * According to the documentation, vswhere is installed in the following fixed location location:
+  * %ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe for 64 bit
+  * %ProgramFiles%\Microsoft Visual Studio\Installer\vswhere.exe for 32 bit
+  * -https://github.com/Microsoft/vswhere/wiki/Installing
+  */
+  let vswherePath = undefined;
+  if(process.env['ProgramFiles(x86)']){
+    vswherePath = Path.join(process.env['ProgramFiles(x86)'], 'Microsoft Visual Studio', 'Installer', 'vswhere.exe');
+  } 
+  if(vswherePath !== undefined && existsSync(vswherePath)){
+    try {
+      const vswhereStdOut = execFileSync(vswherePath, ['-version', '['+String(version)+','+String(version+1)+')', '-products', productId, '-property', 'instanceId'],{timeout: 2000, killSignal: 'SIGKILL'});
```

Comment:
- Instead of `execFileSync` you can utilize `execFile` from `../exec-file` (it returns a Promise). It will make it consistent with other usages across GitHub Desktop
- Perhaps you can use template literal for `-version` argument `['+String(version)+','+String(version+1)+')`
- I'm not sure about this one, but the format of `['+String(version)+','+String(version+1)+')` look a bit odd... it opens with square brackets, and it ends with closing parenthesis?

Reply:
https://github.com/microsoft/vswhere/blob/develop/src/vswhere.lib/VersionRange.cpp#L105-L108 take a look at this to know why the ``)`` is at the end.